src/main/java/edu/umich/its/spe/FileIO.java

```diff
@@ -0,0 +1,118 @@
+package edu.umich.its.spe;
+
+import java.io.File;
+import java.io.IOException;
+
+/*
+ * Implement file IO for the grade input / output.  The implementation must allow configuration
+ * so either get/put or both (or neither) use FileIO.  Any additional properties required for the IO
+ * must either be provided on construction or come from the properties files.
+ *
+ * Methods are static but calls based covers are provided to be consistent with calls
+ * to the ESB implementation.
+ *
+ * Input file format is string version of JSON WAPI wrapped response.  However line breaks are permitted.
+ *
+ *	Throws GradeIOException if there is a problem.
+ *
+ */
+
+import java.util.HashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import edu.umich.ctools.esb.utils.WAPI;
+import edu.umich.ctools.esb.utils.WAPIResultWrapper;
+
+import org.apache.commons.io.FileUtils;
+import org.json.JSONObject;
+
+@Component
+@Qualifier("FileIO")
+
+public class FileIO implements GradeIO {
+	private static final Logger M_log = LoggerFactory.getLogger(FileIO.class);
+
+	/******* Class covers to call underlying static methods *********/
+	@Override
+	public WAPIResultWrapper getGradesVia(SPEProperties speproperties, String gradedAfterTime)
+			throws GradeIOException {
+		return getGradesFileStatic(speproperties,gradedAfterTime);
+	}
+
+	@Override
+	public WAPIResultWrapper putGradeVia(SPEProperties speproperties, HashMap<?, ?> user) {
+		return putGradeFileStatic(speproperties, user);
+	}
+
+	@Override
+	public boolean verifyConnection(SPEProperties speproperties) {
+		return verifyConnectionFileStatic(speproperties);
+	}
+
+	/********** Static methods ************/
+
+	public static WAPIResultWrapper getGradesFileStatic(SPEProperties speproperties, String gradedAfterTime) throws GradeIOException {
+		M_log.debug("getGrades FileIO: properties: {} gradedAfterTime: {}",speproperties,gradedAfterTime);
+		if (speproperties == null || gradedAfterTime == null || gradedAfterTime.length() == 0) {
+			throw new GradeIOException("FileIO Call invalid: speproperties: "+ speproperties+" gradedAfterTime: "+gradedAfterTime);
+		}
+
+		HashMap<String,String> ioProperties = speproperties.getIo();
+		String fileName = SPEUtils.safeGetPropertyValue(ioProperties,"getGradeIO");
+
+		File file = new File(fileName);
```

Comment:
The above 3 lines are repeated below in the `putGradeFileStatic`can take this out as small method and pass in get/putGradeIO as argument